config/v3_0/types/storage.go

```diff
@@ -14,10 +14,46 @@
 
 package types
 
+import (
+	"strings"
+
+	"github.com/coreos/ignition/config/shared/errors"
+	"github.com/coreos/ignition/config/validate/report"
+)
+
 func (s Storage) MergedKeys() map[string]string {
 	return map[string]string{
 		"Directories": "Node",
 		"Files":       "Node",
 		"Links":       "Node",
 	}
 }
+
+func (s Storage) Validate() (r report.Report) {
+	for _, d := range s.Directories {
+		for _, l := range s.Links {
+			if strings.HasPrefix(d.Path, l.Path) {
```

Comment:
This will reject a directory `/a/b/cc` if there's a link `/a/b/c`.  We should probably have tests for those sorts of cases.

Also, we could do better than O(n^2) here.  I don't think that needs to be today, though.

Reply:
hmm given that paths must be clean and we already check for duplicates, we could just append a "/" to the path.

Reply:
Fixed, this also lets us drop the "am I comparing against myself" check in the links bit.